aggregator_api/src/lib.rs

```diff
@@ -0,0 +1,897 @@
+                        )
+                        .await
+                    }
+                    .instrument(info_span!(ENDPOINT_NAME))
+                }
+            })
+            .post("/tasks", {
+                const ENDPOINT_NAME: &str = "post_task";
+                |conn: Conn| {
+                    async move {
+                        endpoint_handler::<C, _>(
+                            conn,
+                            ENDPOINT_NAME,
+                            |aggregator_api, conn| async move {
+                                aggregator_api.handle_post_task(conn).await
+                            },
+                        )
+                        .await
+                    }
+                    .instrument(info_span!(ENDPOINT_NAME))
+                }
+            })
+            .get("/tasks/:task_id", {
+                const ENDPOINT_NAME: &str = "get_task";
+                |conn: Conn| {
+                    async move {
+                        endpoint_handler::<C, _>(
+                            conn,
+                            ENDPOINT_NAME,
+                            |aggregator_api, conn| async move {
+                                aggregator_api.handle_get_task(conn).await
+                            },
+                        )
+                        .await
+                    }
+                    .instrument(info_span!(ENDPOINT_NAME))
+                }
+            })
+            .delete("/tasks/:task_id", {
+                const ENDPOINT_NAME: &str = "delete_task";
+                |conn: Conn| {
+                    async move {
+                        endpoint_handler::<C, _>(
+                            conn,
+                            ENDPOINT_NAME,
+                            |aggregator_api, conn| async move {
+                                aggregator_api.handle_delete_task(conn).await
+                            },
+                        )
+                        .await
+                    }
+                    .instrument(info_span!(ENDPOINT_NAME))
+                }
+            })
+            .get("/tasks/:task_id/metrics", {
+                const ENDPOINT_NAME: &str = "get_task_metrics";
+                |conn: Conn| {
+                    async move {
+                        endpoint_handler::<C, _>(
+                            conn,
+                            ENDPOINT_NAME,
+                            |aggregator_api, conn| async move {
+                                aggregator_api.handle_get_task_metrics(conn).await
+                            },
+                        )
+                        .await
+                    }
+                    .instrument(info_span!(ENDPOINT_NAME))
+                }
+            }),
+    )
+}
+
+async fn endpoint_handler<C: Clock, Fut>(
+    mut conn: Conn,
+    endpoint_name: impl Into<String>,
+    handler: impl Fn(Arc<AggregatorApi<C>>, Conn) -> Fut,
+) -> Conn
+where
+    Fut: Future<Output = Conn>,
```

Comment:
nit: I like to put all the trait bounds either in `<>` or in the `where` clause
```suggestion
async fn endpoint_handler<C: Clock, Fut: Future<Output = Conn>>(
    mut conn: Conn,
    endpoint_name: impl Into<String>,
    handler: impl Fn(Arc<AggregatorApi<C>>, Conn) -> Fut,
) -> Conn
```